src/TropicalGeometry/groebner_basis.jl

```diff
@@ -0,0 +1,129 @@
+###
+# Computing (tropical) Groebner bases in Oscar
+# ============================================
+#
+# For a definition of tropical Groebner basis see Section 2.4 in:
+#   D. Maclagan, B. Sturmfels: Introduction to tropical geometry
+# To see how they can be computed using standard bases see:
+#   T. Markwig, Y. Ren: Computing tropical varieties over fields with valuation
+###
+
+
+
+#=======
+return true if f is homogeneous (w.r.t. total degree)
+return false otherwise
+=======#
+function _is_homogeneous(f::MPolyElem)
+  leadexpv,tailexpvs = Iterators.peel(exponent_vectors(f))
+  d = sum(leadexpv)
+  for tailexpv in tailexpvs
+    if d!=sum(tailexpv)
+      return false
+    end
+  end
+  return true
+end
+
+function has_homogeneous_generators(I::MPolyIdeal{K} where {K})
+  # todo: test whether generators are interreduced
+
+  return all(_is_homogeneous, gens(I))
+end
+export has_homogeneous_generators
+
+
+
+#=======
+tropical Groebner basis
+todo: proper documentation
+Example:
+
+val_2 = TropicalSemiringMap(QQ,2)
+Kx,(x,y,z) = PolynomialRing(QQ,3)
+w = [0,0,0]
+I = ideal([x+2*y,y+2*z])
+groebner_basis(I,val_2,w)
+
+Kt,t = RationalFunctionField(QQ,"t")
+val_t = TropicalSemiringMap(Kt,t)
+Ktx,(x,y,z) = PolynomialRing(Kt,3)
+w = [0,0,0]
+I = ideal([x+t*y,y+t*z])
+groebner_basis(I,val_t,w,return_lead=true)
+=======#
+@doc Markdown.doc"""
+    groebner_basis(I::Ideal, val::TropicalSemiringMap, w::Vector; complete_reduction::Bool, return_lead)
+
+Compute a Groebner basis of `I` over a field with valuation `val` with respect
+to weight vector `w`, that is a finite generating set of `I` whose initial
+forms generate the initial ideal with respect to `w`.
+
+For the definitions of initial form, initial ideal and Groebner basis see
+Section 2.4 of [MS15](@cite).
+
+# Warning
```

Comment:
Consider using an [admonition](https://docs.julialang.org/en/v1/stdlib/Markdown/#Admonitions) like `!!! warning` for cool effect (details in the docs I linked)